mllib/src/main/scala/org/apache/spark/mllib/pmml/export/TreeModelUtils.scala

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.pmml.export
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.collection.mutable.MutableList
+
+import org.dmg.pmml.{Node => PMMLNode, _}
+
+import org.apache.spark.mllib.tree.configuration.{Algo, FeatureType}
+import org.apache.spark.mllib.tree.configuration.Algo._
+import org.apache.spark.mllib.tree.model.{DecisionTreeModel, Node}
+
+private[mllib] object TreeModelUtils {
+
+
+  def toPMMLTree(mllibTreeModel: DecisionTreeModel, modelName: String): TreeModel = {
+
+    val miningFunctionType = getPMMLMiningFunctionType(mllibTreeModel.algo)
+
+    val miningSchema = new MiningSchema()
+      .withMiningFields(getMiningFieldsForTree(mllibTreeModel).asJava)
+
+    val treeModel = new org.dmg.pmml.TreeModel()
+      .withModelName(modelName)
+      .withFunctionName(miningFunctionType)
+      .withNode(buildStub(mllibTreeModel.topNode))
```

Comment:
We could add
.withSplitCharacteristic(SplitCharacteristic.BINARY_SPLIT)
for completeness, as far as I understand MLlib uses a binary split.